debugtools/DDR_VM/src/com/ibm/j9ddr/tools/PointerGenerator.java

```diff
@@ -789,18 +795,21 @@ private void writeEnumEAMethod(PrintWriter writer, String returnType, StructureD
 		writeMethodClose(writer);
 	}
 
-	public static String getOffsetConstant(FieldDescriptor fieldDescriptor) {
-		String fieldName = fieldDescriptor.getName();
-		int index = fieldName.indexOf("_v");
-		if (index == -1) {
-			return fieldName;
+	private String getOffsetConstant(FieldDescriptor fieldDescriptor) {
+		if (opts.get("-l").equals("true")) {
+			return getOffsetConstant(fieldDescriptor.getName());
 		}
+		return fieldDescriptor.getName();
+	}
 
-		if (Character.isDigit(fieldName.charAt(index + 2))) {
-			return fieldName.substring(0, index);
-		} else {
-			return fieldName;
+	private final static Pattern offsetPattern = Pattern.compile("^(.*)_v\\d+$");
```

Comment:
The pattern should require at least one character before `_v` (change `*` to `+`).